python/tank/platform/engine.py

```diff
@@ -207,9 +207,23 @@ def __init__(self, tk, context, engine_instance_name, env):
         # point we want to try and have all app initialization complete.
         self.__run_post_engine_inits()
 
-        if self.name not in [constants.SHELL_ENGINE_NAME, constants.SHOTGUN_ENGINE_NAME] \
-                and self.__has_018_logging_support():
-
+        # The Desktop engine provides its own advanced menu toggle for
+        # debug logging. For the Shotgun engine, we don't want to provide
+        # a debug logging toggle for browser integration. Instead, that
+        # will follow the global debug state, or that managed by the Desktop
+        # engine. For the shell engine, it doesn't make sense to have an
+        # engine command registered for something like toggling debug
+        # logging. If the user wants to control that in the shell, they
+        # either do that with the TK_DEBUG environment variable, or by
+        # setting the state in the LogManager directly if they're running
+        # in tank shell.
+        no_debug_toggle_engines = [
```

Comment:
Why isn't this a property at the engine level? We shouldn't have to edit core to add another engine to the list.

Reply:
Maybe we can introduce the new property now and desktop is updated to use this flag. The constants approach would be left as legacy code paths we can get rid of in the future.